src/vector_index.h

```diff
@@ -32,24 +32,18 @@ typedef enum {
   VECSIM_QT_TOPK,
 } VectorQueryType;
 
-typedef enum {
-  VECSIM_RUN_KNN = 0,
-  VECSIM_RUN_BATCH = 1,
-} VectorQueryRunType;
-
-// This struct always hold parameter name and value as allocated string.
-// First, the parser creates the param, and hold the key-name and value as they appear in the query,
+// This struct holds VecSimRawParam array and bool array.
+// the arrays should have the same length, for testing if the param in some index needs to be evaluated.
+// `params` params will always hold parameter name and value as allocated string.
+// First, the parser creates the param, holds the key-name and value as they appear in the query,
 // and marks if the value is the literal value or an attribute name.
 // Second, in the parameters evaluation step, if a param was marked as an attribute, we try to resolve it,
 // and free its old value and replace it with the actual value if we succeed.
 // It is the VecSim library job to resolve this strings-key-value params (array) into a VecSimQueryParams struct.
 typedef struct {
-  const char *name;
-  size_t namelen;
-  const char *value;
-  size_t vallen;
-  bool isParam;
-} VectorQueryParam;
+  VecSimRawParam *params;
+  bool *isAttr;
```

Comment:
can we rename it to `shouldResolve` or something similar?

Reply:
OK